pkg/util/labels.go

```diff
@@ -92,6 +101,17 @@ func AddObjectLabels(obj runtime.Object, labels labels.Set) error {
 	return nil
 }
 
+// addDeploymentConfigNestedLabels adds new label(s) to a nested labels of a single DeploymentConfig object
+func addDeploymentConfigNestedLabels(obj *deployapi.DeploymentConfig, labels labels.Set) error {
+	if obj.Template.ControllerTemplate.Template.Labels == nil {
+		obj.Template.ControllerTemplate.Template.Labels = make(map[string]string)
+	}
+	if err := MergeInto(obj.Template.ControllerTemplate.Template.Labels, labels, ErrorOnDifferentDstKeyValue); err != nil {
```

Comment:
Does it make sense to use ErrorOnDifferentDstKeyValue? For the other objects in the list we use OverwriteExistingDstKey.